src/dns/ResolvConf.cc

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "base/PackableStream.h"
+#include "dns/ResolvConf.h"
+#include "tools.h"
+
+Dns::ResolvConf &
+Dns::ResolvConf::Current()
+{
+    static ResolvConf instance;
+    return instance;
+}
+
+void
+Dns::ResolvConf::dump(Packable *e)
+{
+    PackableStream out(*e);
+
+    out << "# DNS configuration from " << _PATH_RESCONF << ":" << std::endl;
+
+    if (options.ndots != 0)
+        out << "#    options ndots:" << options.ndots << std::endl;
+
+    if (!search.empty()) {
+        out << "#    search";
+        for (const auto &tld: search)
+            out << " " << tld;
+        out << std::endl;
+    }
+
+    if (!nameservers.empty()) {
+        for (const auto &ns: nameservers)
+            out << "#    nameserver " << ns << std::endl;
+    }
+    out << std::endl;
+}
+
+void
+Dns::ResolvConf::load()
+{
+#if !_SQUID_WINDOWS_
+    search.clear();
+    nameservers.clear();
+    options.clear();
+
+    // default specified for search list
+    if (auto *t = getMyHostname()) {
+        if ((t = strchr(t, '.')))
+            search.emplace_back(SBuf(t+1));
+    }
+
+    FILE *fp = fopen(_PATH_RESCONF, "r");
```

Comment:
Have considered converting this to a std::ifstream and Tokenizer parser. But decided to leave the change relatively simpler to ease review. If others insist that can be included.